src/linker/Linker.Dataflow/Scanner.cs

```diff
@@ -0,0 +1,984 @@
+			switch (GetSourceParameterIndex (thisMethod, operation, out var sourceParameterIndex)) {
+			case SourceParameterKind.Numbered:
+				var targetValue = GetMethodParameterValue (thisMethod, sourceParameterIndex);
+				if (targetValue is MethodParameterValue targetParameterValue)
+					HandleStoreParameter (thisMethod, targetParameterValue, operation, valueToStore);
+				break;
+			// If the targetValue is MethodThisValue do nothing - it should never happen really, and if it does, there's nothing we can track there
+			case SourceParameterKind.This:
+				break;
+			default:
+				break;
+			}
+		}
+
+		private void ScanLdloc (
+			Instruction operation,
+			BlockDataFlowState<MultiValue, ValueSetLatticeWithUnknownValue<SingleValue>> state,
+			MethodDefinition method)
+		{
+			VariableDefinition localDef = GetLocalDef (operation, method.Body.Variables);
+			if (localDef == null) {
+				PushUnknownAndWarnAboutInvalidIL (state, method.Body, operation.Offset);
+				return;
+			}
+
+			bool isByRef = operation.OpCode.Code == Code.Ldloca || operation.OpCode.Code == Code.Ldloca_S;
+
+			MultiValue newSlot;
+			if (isByRef) {
+				newSlot = new MultiValue (new LocalVariableReferenceValue (localDef));
+			} else
+				newSlot = new MultiValue (state.Get (new LocalKey (localDef)));
+			state.Push (newSlot);
+		}
+
+		void ScanLdtoken (Instruction operation, BlockDataFlowState<MultiValue, ValueSetLatticeWithUnknownValue<SingleValue>> state)
+		{
+			switch (operation.Operand) {
+			case GenericParameter genericParameter:
+				var param = new RuntimeTypeHandleForGenericParameterValue (genericParameter);
+				state.Push (new MultiValue (param));
+				return;
+			case TypeReference typeReference when ResolveToTypeDefinition (typeReference) is TypeDefinition resolvedDefinition:
+				// Note that Nullable types without a generic argument (i.e. Nullable<>) will be RuntimeTypeHandleValue / SystemTypeValue
+				if (typeReference is IGenericInstance instance && resolvedDefinition.IsTypeOf (WellKnownType.System_Nullable_T)) {
+					switch (instance.GenericArguments[0]) {
+					case GenericParameter genericParam:
+						var nullableDam = new RuntimeTypeHandleForNullableValueWithDynamicallyAccessedMembers (new TypeProxy (resolvedDefinition),
+							new RuntimeTypeHandleForGenericParameterValue (genericParam));
+						state.Push (new MultiValue (nullableDam));
+						return;
+					case TypeReference underlyingTypeReference when ResolveToTypeDefinition (underlyingTypeReference) is TypeDefinition underlyingType:
+						var nullableType = new RuntimeTypeHandleForNullableSystemTypeValue (new TypeProxy (resolvedDefinition), new SystemTypeValue (underlyingType));
+						state.Push (new MultiValue (nullableType));
+						return;
+					default:
+						PushUnknown (state);
+						return;
+					}
+				} else {
+					var typeHandle = new RuntimeTypeHandleValue (new TypeProxy (resolvedDefinition));
+					state.Push (new MultiValue (typeHandle));
```

Comment:
Nit: I think the `new MultiValue` is unnecessary - it should have an implicit conversion on it.